nx/lib/nx/defn/evaluator.ex

```diff
@@ -235,9 +241,11 @@ defmodule Nx.Defn.Evaluator do
       %{^id => count} when is_integer(count) ->
         {res, [cache | caches]} = eval_apply(op, ans, state, [cache | caches])
         state.gc && :erlang.garbage_collect(self())
+        {res, state} = maybe_debug_node(ans, res, state)
```

Comment:
```suggestion
        {res, state} = debug_node(ans, res, state)
```

I know the intention behind the maybe, but I'd rather just not have it as a prefix -- totally personal preference. I'll apply this change.

Reply:
Also, do we need to return res here? Is it modified?

Reply:
Just checked and it's not. We should also change the return signature

Reply:
Turns out we weren't propagating the updated state. I changed the MapSet to an :ets